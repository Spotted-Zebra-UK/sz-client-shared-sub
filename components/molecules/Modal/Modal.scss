@import '../../../variables';

.Modal {
  background-color: $white;

  &:focus {
    outline: none;
  }
}

.Modal--Desktop {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.Modal--Mobile {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  overflow: auto;
}

.ModalOverlay {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0, 0, 0, 0.54);
}
