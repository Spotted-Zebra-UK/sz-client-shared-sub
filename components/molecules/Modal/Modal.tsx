import './Modal.scss';
import React, { FC } from 'react';
import ReactModal from 'react-modal';

interface IModal {
  onClose: () => void;
  className?: string;
}

const Modal: FC<IModal> = ({ children, onClose, className }) => {
  return (
    <ReactModal
      ariaHideApp={false}
      className={`Modal${className ? ` ${className}` : ''}`}
      isOpen
      onRequestClose={onClose}
      overlayClassName="ModalOverlay"
      shouldCloseOnOverlayClick
    >
      {children}
    </ReactModal>
  );
};

export default Modal;
