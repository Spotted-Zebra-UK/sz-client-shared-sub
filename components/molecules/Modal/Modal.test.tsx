import { shallow } from 'enzyme';
import Modal from './Modal';

describe('Modal', () => {
  let onClose: () => void;

  beforeEach(() => {
    onClose = jest.fn();
  });

  it('should call onClose on modal close', () => {
    const wrapper = shallow(<Modal onClose={onClose} />);

    wrapper.props().onRequestClose();

    expect(onClose).toHaveBeenCalled();
  });

  // it('should render correctly by default', () => {
  //   const wrapper = shallow(
  //     <Modal onClose={onClose}>
  //       <span>Test children</span>
  //     </Modal>
  //   );

  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });

  it('should add className', () => {
    const className = 'test-className';
    const wrapper = shallow(<Modal className={className} onClose={onClose} />);

    expect(wrapper.hasClass(className));
  });
});
