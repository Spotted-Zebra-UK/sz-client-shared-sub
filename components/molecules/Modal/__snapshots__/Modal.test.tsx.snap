// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly by default 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="Modal"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="ModalOverlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <span>
    Test children
  </span>
</Modal>
`;
